tests/HistoryServiceSpec.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module HistoryServiceSpec (spec) where
+
+
+import Control.Concurrent.MVar
+import Control.Monad
+import Control.Monad.IO.Class
+import Control.Monad.Trans.RWS.Strict
+import Data.Aeson (Value(Null))
+import Data.ByteString (ByteString)
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.HashSet as HashSet
+import System.Directory
+import Test.Hspec
+
+import Pact.Server.History.Persistence as DB
+import Pact.Server.History.Service
+import Pact.Server.History.Types
+import Pact.Types.Command
+import Pact.Types.Hash
+import Pact.Types.Server
+import Pact.Types.Term
+
+
+histFile :: FilePath
+histFile = fp ++ "/commands.sqlite"
+
+spec :: Spec
+spec = describe "roundtrip" $ before_ deleteTempFile $ after_ deleteTempFile $ testHistoryDB
+
+deleteTempFile :: IO ()
+deleteTempFile = do
+  isFile <- doesFileExist histFile
+  when isFile $ removeFile histFile
+
+
+dbg :: String -> IO ()
+-- dbg = putStrLn   <- USE THIS TO DEBUG HISTORY STUFF
+dbg = const $ return ()
+
+fp :: FilePath
+fp = "tests/test-log"
+
+cmd :: Command ByteString
+cmd = Command "" [] initialHash
```

Comment:
Probably use something more elaborate than an empty string. This would make the test a bit more robust against field swaps.

Reply:
Since this test doesn't run Pact those code paths don't get exercised, so we'd need to put stuff in there to manually check.

Reply:
Not sure I can follow. I would like to make sure we can detect the case when the wrong text column gets read in by accident.